code/modules/mob/living/carbon/human/species/_species.dm

```diff
@@ -1077,3 +1066,105 @@ It'll return null if the organ doesn't correspond, so include null checks when u
+			playsound(user.loc, 'sound/misc/moist_impact.ogg', 50, TRUE)
+			user.do_attack_animation(target, ATTACK_EFFECT_BITE)
+			target.apply_damage(20, BRUTE, user.zone_selected)
+			if(!target.HasDisease(/datum/disease/zombie))
+				var/datum/disease/zombie/zomb = new /datum/disease/zombie
+				if(target.CanContractDisease(zomb)) // biosuit aint going to protect you buddy
+					target.ForceContractDisease(zomb)
+					target.Dizzy(10 SECONDS)
+					target.Confused(10 SECONDS)
+				else
+					qdel(zomb)
+
+			for(var/datum/disease/zombie/zomb in target.viruses)
+				zomb.stage = max(rand(3, 4), zomb.stage)
+
+			qdel(grabby)
+			return TRUE
+
+		if(HAS_TRAIT(target, TRAIT_I_WANT_BRAINS))
+			to_chat(user, "<span class='warning zombie'>We can't smell any fresh brains in [target].</span>")
+			return
+
+		var/obj/item/organ/internal/brain/eat_brain = target.get_organ_slot("brain")
+		if(!eat_brain || ismachineperson(target))
+			to_chat(user, "<span class='warning zombie'>We can't smell any brains in [target].</span>")
+			return FALSE
+		var/obj/item/organ/external/brain_house = target.get_limb_by_name(eat_brain.parent_organ)
+		if(!brain_house)
+			to_chat(user, "<span class='warning zombie'>We can't smell any brains in [target].</span>")
+			return FALSE
+		if(brain_house.limb_name != user.zone_selected || !brain_house.open)
+			return FALSE
+		if(target.getarmor(brain_house, MELEE) > 0) // dont count negative armor
+			to_chat(user, "<span class='warning zombie'>[target]'s brains are blocked.</span>")
+			return FALSE // Armor blocks zombies trying to eat your brains!
+
+		if(target.getBrainLoss() >= 120)
+			to_chat(user, "<span class='warning zombie'>No more brains left...</span>")
+			return TRUE
+
+		eat_brain.custom_pain("OH GOD!!! THEY'RE EATING MY [uppertext(eat_brain.name)]!!") // gnarly
+		user.visible_message("<span class='danger'>[user] digs their claws into [target]'s [brain_house.name], eating their [eat_brain]!</span>", "<span class='danger zombie'>We feast on [target]'s brains.</span>")
+		if(!target.HasDisease(/datum/disease/zombie))
+			var/datum/disease/zombie/zomb = new /datum/disease/zombie
+			target.ContractDisease(zomb)
+
+		for(var/datum/disease/zombie/zomb in target.viruses)
+			zomb.stage = max(5, zomb.stage)
+
+		if(!do_mob(user, target, 1 SECONDS))
+			return
+
+		playsound(user.loc, 'sound/items/eatfood.ogg', 20, TRUE)
+		playsound(user.loc, 'sound/misc/moist_impact.ogg', 50, TRUE)
+		user.do_attack_animation(target, ATTACK_EFFECT_BITE)
+		target.adjustBrainLoss(5)
+		if(target.stat == CONSCIOUS && prob(33))
+			// Ouch, eaten alive.
+			target.emote("scream")
+
+		if(user.nutrition < NUTRITION_LEVEL_FULL - 10) // no fat zombies
+			user.set_nutrition(user.nutrition + 10)
+		user.heal_overall_damage(2, 2)
+		return TRUE
+
+	//Vampire code
+	var/datum/antagonist/vampire/V = user?.mind?.has_antag_datum(/datum/antagonist/vampire)
+	if(V && !V.draining && user.zone_selected == BODY_ZONE_HEAD)
```

Comment:
```suggestion
	if(!V?.draining && user.zone_selected == BODY_ZONE_HEAD)
```
While we're here

Reply:
I'm not sure these are the same thing logically, and even if it is, I dont like combining ! and ? together because it makes it more uncertain to the reader what could happen

Reply:
They should be, but that's a fair concern